File: problemsets/Codeforces/Go/A835.go
/**
 *
 * Author:  Juarez Paulino(coderemite)
 * Email: juarez.paulino@gmail.com
 *
 */
package main

import "fmt"

func main() {
	var n, a, b, c, d int
	fmt.Scan(&n,&a,&b,&c,&d)
	x := n*a+2*c; y := n*b+2*d
	if x < y {
		fmt.Print("First")
	} else if x > y {
		fmt.Print("Second")
	} else {
		fmt.Print("Friendship")
	}
}

File: problemsets/Codeforces/Python/A835.py
"""
 *
 * Author:  Juarez Paulino(coderemite)
 * Email: juarez.paulino@gmail.com
 *
 """
s,a,b,c,d=map(int,input().split())
e=2*(c-d)+s*(a-b)
print([['Friendship','First'][e<0],'Second'][e>0])